scipy/linalg/flapack_pos_def_tri.pyf.src

```diff
@@ -12,3 +12,73 @@ subroutine <prefix>ptsv(n, nrhs, d, e, b, info)
     integer intent(out) :: info
 
 end subroutine <prefix>ptsv
+
+
+subroutine <prefix>pttrf(n, d, e, info)
+    ! d, e, info  = pttrf(d, e, overwrite_d=0, overwrite_e=0)
+    !
+    ! ?PTTRF computes the L*D*L**T factorization of a symmetric positive
+    ! definite tridiagonal matrix A.  The factorization may also be regarded as ! having the form A = U**T*D*U.
+
+    callstatement (*f2py_func)(&n, d, e, &info)
+    callprotoargument int*, <ctypereal>*, <ctype>*, int*
+
+    integer intent(hide), depend(d), check(n > 1) :: n = len(d)
+    <ftypereal> intent(in, out, copy), dimension(n) :: d
+    <ftype> intent(in, out, copy), dimension(n - 1), depend(n) :: e
```

Comment:
Connected to the check I mentioned above you can do this via `dimension((n>0?n-1:0))`. Then you don't need to care what `n` is.

Reply:
Line is now `<ftype> intent(in, out, copy), dimension((n>0?n-1:0)) :: e`

Reply:
(By "you don't need to care what `n` is" you mean I can eliminate `depend(n)`, right?)

Reply:
No that we still need. But we don't need to worry about n being zero and we are left with `dimension(-1)` which is a runtime error regardless.

Reply:
Added back.